### PR TITLE
Update OpenTelemetry to 1.30.1

### PR DESCRIPTION
**Pull request by ebyhr**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DriverConnectionFactory.java [hunk 1]
```diff
@@ -75,9 +75,7 @@ public Connection openConnection(ConnectorSession session)
             throws SQLException
     {
         Properties properties = getCredentialProperties(session.getIdentity());
-        // TODO: Telemetry is disabled due to NPE being thrown on null connection
-        // Connection connection = dataSource.getConnection(properties);
-        Connection connection = driver.connect(connectionUrl, properties);
+        Connection connection = dataSource.getConnection(properties);
```
Comment (ebyhr) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DriverConnectionFactory.java:78]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.30.0 contains 
> Return default DbInfo when connection is null (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9413)

File: lib/trino-filesystem/src/main/java/io/trino/filesystem/tracing/Tracing.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.api.trace.StatusCode;
-import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
```
Comment (ebyhr) [inline lib/trino-filesystem/src/main/java/io/trino/filesystem/tracing/Tracing.java:20]: `io.opentelemetry.semconv.trace.attributes.SemanticAttributes` is deprecated.

**Review by wendigo — APPROVED** [review]

**Comment by ebyhr** [comment]
@wendigo Sorry, I was missing your #19075

**Comment by wendigo** [comment]
No problem @ebyhr

**Pull request merged**